src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java

```diff
@@ -126,14 +126,12 @@ private Organisation saveOrganisation(Organisation organisation) {
         return persistedOrganisation;
     }
 
-    private void addPbaAccountToOrganisation(
-            List<String> paymentAccounts,
-            Organisation organisation) {
+    public void addPbaAccountToOrganisation(List<String> paymentAccounts, Organisation organisation) {
 
         if (paymentAccounts != null) {
             paymentAccounts.forEach(pbaAccount -> {
                 if (pbaAccount == null || !pbaAccount.matches("(PBA|pba).*") || !pbaAccount.matches("^[a-zA-Z0-9]+$")) {
```

Comment:
You are overcomplicating this. You should be using paymentAccounts.filter(pbaAccout -> /*add the contents of your if statement here*/) then you can throw something. I would recommend separating out the concerns though and you can then use .map to change from a list of pbaAccounts to a list of String (if there are any). You can than perform .collect(Collectors.joining(",")) to concatenate the list into a comma separated String and use a private method to check whether that errorMessage.isEmpty() is true if so it will do nothing. If not, then the private method (or better for testing purposes rather than a private method a  re-usable a public method in an validateWithExceptionService, as was done in UserProfile) would then throw the exception and instead of having the first failed PBA account it would have a record of all of the failed PBA accounts if there were multiple ones, also it would obey the single responsibility pattern, also it would be easier to test.